packages/core/src/deployment.ts

```diff
@@ -97,8 +109,13 @@ export async function waitAndValidateDeployment(provider: EthereumProvider, depl
 }
 
 export class TransactionMinedTimeout extends Error {
-  constructor(readonly deployment: Deployment) {
-    super(`Timed out waiting for transaction ${deployment.txHash}`);
+  constructor(readonly deployment: Deployment, configurableTimeout: boolean) {
+    super(
+      `Timed out waiting for transaction ${deployment.txHash}. Run the function again to continue waiting for the transaction confirmation.` +
+        (configurableTimeout
+          ? ` If the problem persists, adjust the polling parameters with the timeout and pollingInterval options.`
+          : ``),
```

Comment:
Can this class extend `UpgradesError` instead, so we can use the second error line for the suggestions:

```suggestion
      `Timed out waiting for transaction ${deployment.txHash}.`,
      () => `Run the function again to continue waiting for the transaction confirmation.` +
        (configurableTimeout
          ? `\nIf the problem persists, adjust the polling parameters with the timeout and pollingInterval options.`
          : ``),
```